app/src/main/java/com/github/giommok/softwaredevproject/FirebaseAccount.java

```diff
@@ -268,7 +277,31 @@ private static void syncGetDiscoveredHeight(DataSnapshot discoveredHeightsFromHa
         for (Map.Entry<String, HashMap<String, String>> entry : entries.entrySet()){
             Object value = entry.getValue();
             long retrievedVal = ((ArrayList<Long>) value).get(0);
-             discoveredPeakHeights.add((int)retrievedVal);
+            discoveredPeakHeights.add((int)retrievedVal);
+        }
+    }
+
+    private static void syncFriendsFromProfile(DataSnapshot parent) {
+        ArrayList<FriendItem> tempFriends = new ArrayList<>();
```

Comment:
This step could be avoided if you use a `hashSet`

Reply:
Discussed on Discord.